src/main/java/entity/Portfolio.java

```diff
@@ -0,0 +1,49 @@
+package entity;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Portfolio {
+
+    private final Map<String, UserStock> stocks;
+
+    public Portfolio() {
+        this.stocks = new HashMap<>();
+    }
+
+    // Constructor that initializes Portfolio with a map of tickers and their respective quantities
+    public Portfolio(Map<String, Integer> stocks) {
+        this.stocks = new HashMap<>();
+        StockMarket stockMarket = StockMarket.Instance();
+
+        stocks.forEach((ticker, quantity) -> {
+            Stock stock = stockMarket.getStock(ticker);
+            if (stock != null) {
+                double price = stock.getPrice();
+                if (price > 0) {
+                    this.stocks.put(ticker, new UserStock(stock, price, quantity));
+                } else {
+                    System.err.println("Invalid price for stock: " + ticker);
```

Comment:
Instead of printing these exception messages in the console, we may want to throw an Exception that can be captured by the Interactor and displayed to the view